src/NServiceBus.Core/Pipeline/Incoming/DeserializeMessageConnector.cs

```diff
@@ -120,18 +126,16 @@ LogicalMessage[] Extract(IncomingMessage physicalMessage)
             return logicalMessages;
         }
 
-        static bool DoesTypeHaveImplAddedByVersion3(string existingTypeString)
-        {
-            return existingTypeString.Contains("__impl");
-        }
+        static bool DoesTypeHaveImplAddedByVersion3(string existingTypeString) => existingTypeString.AsSpan().IndexOf("__impl".AsSpan()) != -1;
```

Comment:
This change alone is roughly 3% gain. I can share the results tomorrow

Reply:
![image](https://user-images.githubusercontent.com/174258/160104715-705af02e-9406-4e04-91e3-15ec7cc90632.png)

Reply:
Does `"__impl".AsSpan()` get compiled down to a constant? Would introducing that span as a static readonly improve things further?

Reply:
There are all sorts of magic happening there by the compiler that makes it work. You can store a reference to a span on the class level. That is actively prevented by the compiler.